src/Drupal/Driver/Fields/Drupal8/LinkHandler.php

```diff
@@ -13,18 +13,34 @@ class LinkHandler extends AbstractHandler {
   public function expand($values) {
     $return = array();
     foreach ($values as $value) {
+      $uri = $value[1];
+
+      // If the uri has no scheme (and is not protocol relative) attempt to find
+      // a node with that label.
+      if (empty($uri_parts['scheme']) && strpos($uri, '//') !== 0) {
+        $entity_type_id = 'node';
+        $query = \Drupal::entityQuery($entity_type_id)->condition('title', $uri);
+        $entities = $query->execute();
+        if (!empty($entities)) {
+          $nid = array_shift($entities);
+          $uri = 'entity:' . $entity_type_id . '/' . $nid;
+        }
+      }
```

Comment:
I agree! This is bringing the LinkHandler in line with what the EntityReferenceHandler does which let's you reference entities by their label.

Using internal URIs would work with the current implementation however how would you know what the entity_id is of an entity you just created in your behat test?